# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

#### My Lane: Refresh/Content Opportunity Scoring

##### Why? I chose this lane because it mainly focuses on feature engineering, ranking and designs, which are core concepts that are important in machine learning.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

What decision does this work improve?
This model helps clients make better decisions on what to prioritize when trying to increase their ranking and traffic on the web.

Who acts on it?
The clients content/SEO team, i.e. The people responsible for maintaining and improving the website content

Cost of a wrong call?
The model could flag a particular page that does not need optimization and this could lead to wasted time and efforts by the team. It could also lead to the team second-guessing the system, which could lead to them dropping it.



## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import pandas as pd
df = pd.read_csv('C:\\Users\\hamto\\OneDrive\\Desktop\\Flyrank-Machine-Learning-Internship\\data\\raw\\content_refresh_anonymized.csv')
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [11]:
print(df.shape)
print(df.columns.tolist())

(30000, 44)
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [12]:
# near miss zone is defined as positions 8-20
near_miss_pct = df['avg_position'].between(8, 20).mean() * 100
print(f"{near_miss_pct:.1f}% of content sits in the near-miss zone (position 8-20)")

34.6% of content sits in the near-miss zone (position 8-20)


In [13]:
# decay
declining_pct = (df['trend_direction'] == 'down').mean() * 100
print(f"{declining_pct:.1f}% of content is declining in performance")

54.2% of content is declining in performance


### What the numbers tell us
This tells us that about 34.6% of content is one push away from page 1, and about 54.2% of content is declining in performance . This is a large population that a scoring model like this can prioritize.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What This Work Can and Cannot Claim

**What it will say: observed**
This analysis reports what has already happened in the data — actual positions, actual clicks, actual trends over the observed window. When we say "38% of content sits in positions 8–20," that is a description of the current state.

**What it will say: directional**
Where we identify patterns — declining trend, stale content, near-miss positioning — we are pointing at *direction of movement*, not precise magnitude of future outcome. 
"This page is losing ground" is a directional claim. "This page will gain exactly 6 positions if refreshed" is not something this work can support.

**What it will say: decision-support**
The output is a prioritized queue with reasons attached — designed to help a human content team decide where to spend limited time next. It surfaces candidates and evidence; it does not make the final call. The team's judgment, resourcing, and execution quality remain the deciding factors in outcome.

---

**What it will never say: causal proof**
Correlation between a feature (e.g., content age, word count, freshness) and performance is not evidence that changing that feature *causes* a performance change. Pages with strong signals may simply come from stronger domains, better topics, or better original writing — confounds this analysis does not control for. 
We are not running a controlled experiment (e.g., an A/B test on refreshed vs. non-refreshed pages), so no causal claim ("refreshing X caused Y% traffic lift") 
is supportable from this data alone.

**What it will never say: "predicting Google"**
This work does not model or claim to predict Google's ranking algorithm, its weighting of signals, or future algorithm changes. Google's ranking system is a black box with hundreds of undisclosed, frequently-updated factors. Any pattern found here is a *correlation observed in this client's historical data*, not a discovered ranking rule, and it may not generalize to other sites, other verticals, or even the same site after the next algorithm update.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.